### PR TITLE
Feature/stongly-typed-navigation

### PR DESCRIPTION
**Pull request by Eimen2018**
- Adds strongly typed navigation 
- Adds custom transition option for custom routes

### CODE REVIEW

File: packages/stacked/example/lib/app/app.dart [hunk 1]
```diff
@@ -19,7 +19,7 @@ import 'package:stacked_themes/stacked_themes.dart';
     MaterialRoute(page: HomeView, initial: true),
     MaterialRoute(page: BottomNavExample),
     MaterialRoute(page: StreamCounterView),
-    CupertinoRoute<Map<String, List<String>>>(page: DetailsView),
+    CupertinoRoute<Map<String, List<String>>?>(page: DetailsView),
```
Comment (eb-so) [inline packages/stacked/example/lib/app/app.dart:22]: To activate the type inference we have to add it here right?

If the above is right what happen when we change one of the class parameters without adjusting the type here?

File: packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart [hunk 2]
```diff
@@ -15,7 +15,7 @@ class StackedLocatorContentGenerator extends BaseGenerator {
   });
   @override
   String generate() {
-    writeLine("// ignore_for_file: public_member_api_docs");
+    writeLine("// ignore_for_file: public_member_api_docs, unused_import");
```
Comment (eb-so) [inline packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart]: Why did you add ", unused_import"

File: packages/stacked_services/lib/src/navigation/navigation_service.dart [hunk 3]
```diff
@@ -187,10 +187,11 @@ class NavigationService {
     int? id,
     bool preventDuplicates = true,
     Map<String, String>? parameters,
+    RouteTransitionsBuilder? transition,
```
Comment (eb-so) [inline packages/stacked_services/lib/src/navigation/navigation_service.dart:190]: I think this is a fix for adding transtion to navigation with named route right?

File: packages/stacked/example/lib/app/app.dart (hunk 1 above)
Comment (Eimen2018) [reply packages/stacked/example/lib/app/app.dart:22]: This is the expected return type when navigating back. if the user presses the back button it would be null causing a null issue. that's why i added the question mark... to indicate it might be null

File: packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart (hunk 2 above)
Comment (Eimen2018) [reply packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart]: I added it before the imports was fixed I will remove it

File: packages/stacked_services/lib/src/navigation/navigation_service.dart (hunk 3 above)
Comment (Eimen2018) [reply packages/stacked_services/lib/src/navigation/navigation_service.dart:190]: Yup! 👍

**Comment by eb-so** [comment]
@Eimen2018 Awesome work Ayman you completed this feature in a short period of time and also solved another issue"Custom transitions when using named routes." which is Great 🎉 
My only request is to create a new PR for the "Custom transitions when using named routes." feature and merge it first then you can complete your work on the strong type navigation

**Review by FilledStacks — APPROVED** [review]
The implementation looks good! Get those Unit tests fixed up and this will be ready to push out.

**Comment by FilledStacks** [comment]
I'm approving this and merging it int. There's been no objections.
